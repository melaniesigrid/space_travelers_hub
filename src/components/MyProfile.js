import React from 'react';
import { useSelector } from 'react-redux';
import { Container, Table } from 'react-bootstrap';

function MyProfile() {
  const missions = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((mission) => mission.reserved);
  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <div className="ProfileContent">
      <Container className="d-flex">
        <Container>
          <Table striped bordered hover>
            <thead>
              <tr>
                <th>My Missions</th>
              </tr>
            </thead>
            <tbody>
              {
                reservedMissions.map((mission) => (
                  <tr key={mission.mission_id}>
                    <td>{mission.mission_name}</td>
                  </tr>
                ))
              }
            </tbody>
          </Table>
        </Container>
        <Container>
          <Table striped bordered hover>
            <thead>
              <tr>
                <th>My Rockets</th>
              </tr>
            </thead>
            <tbody>
              {
                reservedRockets.map((rocket) => (
                  <tr key={rocket.id}>
                    <td>{rocket.rocketName}</td>
                  </tr>
                ))
              }
            </tbody>
          </Table>
        </Container>
      </Container>
    </div>
  );
}

export default MyProfile;
