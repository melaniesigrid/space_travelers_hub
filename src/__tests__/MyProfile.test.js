import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile';

it('MyProfile Component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
