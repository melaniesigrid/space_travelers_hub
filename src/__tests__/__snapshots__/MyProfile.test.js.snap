// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component renders correctly 1`] = `
<div
  className="ProfileContent"
>
  <div
    className="d-flex container"
  >
    <div
      className="container"
    >
      <table
        className="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              My Missions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      className="container"
    >
      <table
        className="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              My Rockets
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
